src/content/blog/legacy-projects-with-unsupported-php-and-mysql-using-ddev.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "Using DDEV to spin up a legacy PHP application"
+pubDate: 2025-05-27
+modifiedDate: 2025-05-27
+summary: How to use DDEV with outdated PHP and MySQL versions
+author: Garvin Hicking
+featureImage:
+  src: /img/blog/2025/05/museums-victoria-Di7WfLcrJ_I-unsplash.jpg
+  alt: Legacy computer museum
+  credit: "Photo by [Museums Victoria](https://unsplash.com/@museumsvictoria) on [Unsplash](https://unsplash.com/photos/gray-mechanical-machine-lot-beside-wall-Di7WfLcrJ_I)"
+categories:
+  - Guides
+---
+
+_This guest post is by DDEV community member and [TYPO3](https://typo3.org) contributor [Garvin Hicking](/blog/author/garvin-hicking/)._
+
+In my daily work, I develop [TYPO3](https://typo3.org)-based projects and also contribute to the TYPO3 CMS OpenSource project itself.
+
+Usually this means working with actively supported and up-to-date PHP versions as well as database systems like MySQL/PostgreSQL/MariaDB.
+
+Just recently I had to migrate a very outdated project: TYPO3 4.5, which utilized MySQL 5.5 and PHP 5.3. When that project was initially developed, it was done with XAMPP and later Vagrant-based VMs. This has been long superseded with using Docker and specifically DDEV for ease-of-use.
+
+So naturally I wanted to be able to use DDEV for the legacy project to get it working just as it is running on the (outdated) hosting provider's shared web servers.
+
+I quickly faced three major issues:
+
+- No PHP 5.3 out-of-the-box support from DDEV; it starts with 5.6 as of the time of this writing
+- No MySQL 5.5 ARM64 support either; it starts with 5.7
+- Additionally, I use an Apple MacBook Pro M1 with ARM-chipset, which has no "official" MySQL 5.5 support
+
+Thanks to the outstanding DDEV support on Discord, I was quickly able to find a way with minimal effort, just by creating very small custom, additional docker-compose YAML files.
+
+One advantage (of many) of using DDEV instead the underlying Docker Compose is that so many things are pre-configured and "just work". So I really did not want to migrate everything to Docker Compose on my own, do my custom routing, PHP-FPM integration and whatnot.
+
+Just being able to "bait and switch" the PHP and DB container with a different base Docker image was all that was needed for me:
+
+## Step 1: Base config
+
+I created the base `~/legacyphp/.ddev/config.yaml` file manually inside my `~/legacyphp` project directory, setting `legacyphp` as the project name.
+
+Note that I configured PHP and MySQL versions that are supported by DDEV for this first:
+
+```yaml
+name: legacyphp
+type: php
+docroot: htdocs
+php_version: "8.3"
+webserver_type: apache-fpm
+database:
+  type: mysql
+  version: "8.0"
+```
+
+## Step 2: Rewire DB
+
+Next I created the very small file `~/legacyphp/.ddev/docker-compose.db.yaml` in the same directory next to `config.yaml`:
+
+```yaml
+services:
+  db:
+    platform: linux/amd64
+    build:
+      args:
+        BASE_IMAGE: ddev/ddev-dbserver-mysql-5.5:v1.24.6
+```
```

Comment:
Trick DDEV into thinking it's MySQL 8.0:

````suggestion
    entrypoint: ["sh", "-c", "cp /docker-entrypoint.sh ~/docker-entrypoint.sh; sed -i '157s/.*/if false; then/' ~/docker-entrypoint.sh; sed -i '175s|.*|echo 'mysql_8.0' >/var/lib/mysql/db_mariadb_version.txt|' ~/docker-entrypoint.sh; exec ~/docker-entrypoint.sh"]
```
````

Reply:
This is the file that was edited https://github.com/ddev/ddev/blob/main/containers/ddev-dbserver/files/docker-entrypoint.sh